src/backup.ts

```diff
@@ -0,0 +1,387 @@
+import {EnumKey, RequestConfig, TranslateEnumKeys} from '.';
+import {Metadata, Operation as GaxOperation} from 'google-gax';
+import * as extend from 'extend';
+import {DateStruct, PreciseDate} from '@google-cloud/precise-date';
+import {ServiceError, status} from 'grpc';
+
+// Like LongRunningCallback<Backup> but with more specific type for operation parameter
+export interface CreateBackupCallback {
+  (
+    err: ServiceError | null,
+    resource?: Backup | null,
+    // More specific type for CreateBackup operation
+    operation?: CreateBackupGaxOperation | null,
+    apiResponse?: IOperation
+  ): void;
+}
+
+export interface CreateBackupGaxOperation extends GaxOperation {
+  // Overridden with more specific type for CreateBackup operation
+  metadata: Metadata &
+    databaseAdmin.spanner.admin.database.v1.ICreateBackupMetadata;
+}
+
+export type CreateBackupResponse = [
+  Backup,
+  CreateBackupGaxOperation,
+  databaseAdmin.longrunning.IOperation
+];
+
+/**
+ * IBackup structure with backup state enum translated to string form.
+ */
+type IBackupTranslatedEnum = TranslateEnumKeys<
+  databaseAdmin.spanner.admin.database.v1.IBackup,
+  'state',
+  typeof databaseAdmin.spanner.admin.database.v1.Backup.State
+>;
+
+export type GetMetadataResponse = [IBackupTranslatedEnum];
+type GetMetadataCallback = RequestCallback<IBackupTranslatedEnum>;
+
+type UpdateExpireTimeCallback = RequestCallback<Backup>;
+
+type DeleteCallback = RequestCallback<void>;
+
+/**
+ * The {@link Backup} class represents a Cloud Spanner backup.
+ *
+ * Create a `Backup` object to interact with or create a Cloud Spanner backup.
+ *
+ * @class
+ *
+ * @example
+ * const {Spanner} = require('@google-cloud/spanner');
+ * const spanner = new Spanner();
+ * const instance = spanner.instance('my-instance');
+ * const backup = instance.backup('my-backup');
+ */
+class Backup {
+  request: <T, R = void>(
+    config: RequestConfig,
+    callback: RequestCallback<T, R>
+  ) => void;
+
+  formattedName_: string;
+
+  constructor(
+    private instance: Instance,
+    private backupId: string,
+    private databasePath?: string,
+    private expireTime?: PreciseDate
+  ) {
+    this.request = instance.request;
+    this.formattedName_ =
+      this.instance.formattedName_ + '/backups/' + this.backupId;
+  }
+
+  create(): Promise<CreateBackupResponse>;
+  create(callback: CreateBackupCallback): void;
+  /**
+   * Create a backup.
+   *
+   * @method Backup#create
+   * @returns {Promise<CreateBackupResponse>} when resolved, the backup
+   *     operation will have started, but will not have necessarily completed.
+   *
+   * @example
+   * const {Spanner} = require('@google-cloud/spanner');
+   * const spanner = new Spanner();
+   * const instance = spanner.instance('my-instance');
+   * const database = spanner.database('my-database');
+   * const oneDay = 1000 * 60 * 60 * 24;
+   * const ExpiryTime = new PreciseDate(Date.now() + oneDay)
```

Comment:
```diff
- ExpiryTime
+ expiryTime
```

Reply:
+ trailing semicolon at the end of the line :)

Reply:
Oh yikes, sorry :(